openedx/core/djangoapps/video_module/config/models.py

```diff
@@ -0,0 +1,70 @@
+"""
+Configuration models for Video XModule
+"""
+from config_models.models import ConfigurationModel
+from django.db.models import BooleanField
+
+from openedx.core.djangoapps.xmodule_django.models import CourseKeyField
+from request_cache.middleware import request_cached
+
+
+class HLSPlaybackEnabledFlag(ConfigurationModel):
+    """
+    Enables HLS Playback across the platform.
+    When this feature flag is set to true, individual courses
+    must also have HLS Playback enabled for this feature to
+    take effect.
+    """
+    # this field overrides course-specific settings
+    enabled_for_all_courses = BooleanField(default=False)
+
+    @classmethod
+    @request_cached
+    def feature_enabled(cls, course_id):
+        """
+        Looks at the currently active configuration model to determine whether
+        the HLS Playback feature is available.
+
+        If the feature flag is not enabled, the feature is not available.
+        If the flag is enabled for all the courses, feature is available.
+        If the flag is enabled and the provided course_id is for an course
+            with HLS Playback enabled, the feature is available.
+
+        Arguments:
+            course_id (CourseKey): course id for whom feature will be checked.
+        """
+        if not HLSPlaybackEnabledFlag.is_enabled():
+            return False
+        elif not HLSPlaybackEnabledFlag.current().enabled_for_all_courses and course_id:
```

Comment:
Why we need course_id in this if? Is there a scenario where course_id could be False/None? I don't think we need `course_id` here.